### ISSUE TITLE
Buttons do not work on native Home Assistant app for macOS

### ISSUE THREAD

**Issue by JiDoan** [Nerwyn/android-tv-card#70]
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser

**Release with the issue:**
3.5.0

**Last working release (if known):**
Unsure

**Browser and Operating System:**
Native macOS app 2024.2, macOS 14.4

**Description of problem:**
The buttons on the remote no longer work on the native Home Assistant companion app that is downloaded from the App Store for macOS. The same remote works on any other browser or on iOS. The buttons used to work at some point in time but I do not know which version broke the remote. I think the companion app update broke it and not an update to android-tv-card, but I cannot confirm.

**Javascript errors shown in the web inspector (if applicable):**
Unable to open web inspector with companion app.

**Your Full Remote Config**

```yaml
type: custom:android-tv-card
remote_id: remote.living_room_tv
media_player_id: media_player.living_room_tv
keyboard_id: media_player.android_tv_sony
alt_volume_icons: true
rows:
  - - power
    - null
    - search
    - keyboard
  - - youtube
    - max
    - plex_alt
  - navigation_buttons
  - - back
    - home
    - tv
  - - volume_buttons
custom_sources:
  plex_alt:
    icon: phu:plex
    source: plex://

```

**Additional information:** 
N/A

**Comment by Nerwyn** [comment]
Your config is working on my Google TV. It could be an issue with the card deciding on whether to use touch vs mouse handlers. Does your device have a touch screen? Does it work for touch vs mouse clicks? I should look into reworking the mouse vs touch logic to include both handlers and prevent default on the touch events.

**Comment by JiDoan** [comment]
My device is a MacBook so unfortunately no touch screen to test with. I think it has something to do with the Mac app since the same config works with the browser on the same device or on iOS with a touch screen. Not sure how the app is translating the clicks into the embedded web view...

**Comment by Nerwyn** [comment]
I dug out my old work MacBook and can confirm that something is not working with the MacOS Home Assistant app. Reading more about the MacOS HA app, it looks like it's a port of the iOS app so this being a problem with the touch handlers is definitely a possibility.

3.5.1-alpha.002 fixes buttons by binding both mouse and touch handlers to buttons all the time, which makes me think my assumption was correct. This is a bug in the MacOS HA app which causes touch events to be available in the app window but not actually usable since it isn't a touch screen. BUT this alpha also breaks touchscreen scrolling when the user starts on a button so it isn't viable yet. I need to research how other custom cards and Home Assistant handles this.

**Comment by Nerwyn** [comment]
Wait now touch scroll is working? Maybe it was a caching issue? I'm going to further work on this solution and will update when I have a beta for you to try.

**Comment by Nerwyn** [comment]
@JiDoan can you try the 3.5.1 beta and test it on all your devices to make sure it's working for you? It's working for me on Windows Brave and Firefox, the Android app and Firefox, the iPhone app, and the MacOS app. I'll also have to fork these fixes to my service call tile feature project since the same issue is present in that.

**Comment by JiDoan** [comment]
How do you test scrolling? The buttons appear to work on macOS though

**Comment by Nerwyn** [comment]
Scrolling on touchscreens. Using `preventDefault()` on touch events breaks touch scrolling, so it isn't a viable solution. It also isn't possible to call `preventDefault()` on passive event listeners so that was a bad solution anyway (thanks for nothing Stack Overflow).

I have a different idea for a solution for ignoring mouse events if a touch event is triggered and will try implementing it tomorrow. As it is now the beta just throws errors on the touch events and triggers mouse events instead, which is why it is now working on the MacOS HA app.

**Comment by JiDoan** [comment]
It seems to work for me as expected on iOS, touchpad and button scrolling as you described. Does it have issues on android?

https://github.com/Nerwyn/android-tv-card/assets/2526517/450bd242-1f28-4fb0-a526-65ceac5d407b

**Comment by Nerwyn** [comment]
Is that the latest alpha or the beta? If it's the alpha, great! If it's the beta then it's working because the touch handlers are failing and the mouse handlers are being called, which is two wrongs making a right.

**Comment by Nerwyn** [comment]
Good news it seems to be working well on all platforms I've tried. I'm going to work on a few more tweaks and then mark it as a beta, and then work on porting this code over to my tile feature project.

**Comment by Nerwyn** [comment]
Alright, beta 3 release candidate 1 is up. Let me know how it works on all your devices! It looks like it's working great on mine, but I'm cautious since it includes a significant logic change to how events are handled.

**Comment by JiDoan** [comment]
Thanks! Beta 3 seems to work well on macOS app, a chromium browser on both macOS and Windows, and iOS. Scrolling also works as expected!

**Comment by Nerwyn** [comment]
I thought it was working fine on Android but it's  still firing both events. Ugh.

**Comment by Nerwyn** [comment]
And now it's not after clearing cache. I'm going to make a small change for beta 4 and keep using it over the next few days to make sure it's fixed for good.

**Comment by Nerwyn** [comment]
Ended up running into and fixing a multi-touch issue today so now it's on beta 5.

**Comment by JiDoan** [comment]
Thanks for your help on this! I'll give it a try and let you know if I run into any issues!

**Comment by Nerwyn** [comment]
I think it's fully working as of beta 5 and am going to release it. Let me know if you run into any issues! And thanks for your help.

**Comment by Nerwyn** [comment]
Fixed in [3.5.1](https://github.com/Nerwyn/android-tv-card/releases/tag/3.5.1)
